contrib/backporting/check-stable

```diff
@@ -152,7 +153,8 @@ generate_commit_list_for_pr () {
     entry_id=${entry_array[0]}
     entry_sha=${entry_array[1]}
     entry_sub=${entry_array[@]:2}
-    upstream="$(git log -F --since="1year" --pretty="%H" --no-merges --extended-regexp --grep "^$entry_sub" $REMOTE/master)"
+    entry_sub_re="^$(sed 's/[.^$*+?()[{\|]/\\&/g' <<< "$entry_sub")" # adds backslashes for extended regex
+    upstream="$(git log -F --since="1year" --pretty="%H" --no-merges --extended-regexp --grep "$entry_sub_re" $REMOTE/master)"
```

Comment:
Not sure why we have both `-F` and `--extended-regexp`. From the manpage:
```Roff
-E, --extended-regexp
    Consider the limiting patterns to be extended regular expressions instead of the default basic regular expressions.

-F, --fixed-strings
    Consider the limiting patterns to be fixed strings (don’t interpret pattern as a regular expression).
```
I'm guessing the second overwrites the first.

Reply:
Not sure it's worth it, but if we wanted to list commits from a PR without risk of false positives, we could:
1. List all commits on master containing exactly `$entry_sub` with `-F` (no regex).
2. Filter commits that are not part of the PR being considered by listing PRs associated with each commit. A [recent addition to the GitHub API](https://developer.github.com/v3/repos/commits/#list-pull-requests-associated-with-a-commit) enables that. You can try `curl -H 'Accept: application/vnd.github.groot-preview+json' https://api.github.com/repos/cilium/cilium/commits/7b669957f933fdec13d033656e5cc5edaf695996/pulls` to try it on the latest commit on master, 7b669957f933fdec13d033656e5cc5edaf695996.

Reply:
Good catch! I actually did not notice the `-F` there, as I did not realize it was an alias for `--fixed-strings` as I was aware of `--fixed-strings`, but decided against it because of the false positive problem (i.e. it will e.g. match any commit which contains a reference via `Fixes <sha> ("<subject>")`.

I'll remove the `-F` from the PR. The usage of the GitHub API to avoid false positives is probably a bigger change. If it makes the `git patch-id` unnecessary (which also cases problems), I think it might be worth a try though, I'm not too familiar with the scripts yet to judge that.